src/slack/api.cr

```diff
@@ -38,6 +42,16 @@ class Slack::API
     raise Error.new(error.not_nil!)
   end
 
+  private def post(url, params : Hash(String, String))
+    response = @client.post "#{url}?token=#{@token}#{encode_params(params)}"
+    handle(response) {}
+  end
+
+  private def encode_params(params : Hash(String, String))
```

Comment:
Check [HTTP::Params#build](https://crystal-lang.org/api/0.18.7/HTTP/Params.html#build%28%26block%29%3AString-class-method)

You can probably build the query string in one pass with `@token`, and what's in params. And you can remove the restriction in `params` (it will be helpful for the change I'll comment a few lines above)